scraping/spiders/umapa.py

```diff
@@ -0,0 +1,104 @@
+import logging
+
+import scrapy
+
+from config.constants import DEFAULT_CATEGORY
+
+
+# Ref: https://www.mapotic.com/api/v1/maps/10392/public-categories/
+CATEGORIES = {
+    "Education": "Education",
+    "Psychological help": "Mental help",
+    "Legal services": DEFAULT_CATEGORY,
+    "Translators": DEFAULT_CATEGORY,
+    "Integration": DEFAULT_CATEGORY,
+    "Food": "Food",
+    "Institutions": DEFAULT_CATEGORY,
+    "Leisure activities": DEFAULT_CATEGORY,
+    "Doctors": "Medical",
+    "Transport": "Transport",
+    "Other": DEFAULT_CATEGORY,
+}
+
+HEADERS = {
+    "Accept": "application/json",
+}
+
+
+log = logging.getLogger(__name__)
+
+
+class UmapaSpider(scrapy.Spider):
+    """
+    Umapa crawler.
+    """
+
+    name = "umapa"
+    allowed_domains = ["umapa.eu", "www.mapotic.com"]
+    start_urls = [
+        "https://www.mapotic.com/api/v1/maps/10392/pois.geojson/?",
+    ]
+
+    def start_requests(self):
+        for url in self.start_urls:
+            yield scrapy.Request(url=url, callback=self.parse, headers=HEADERS)
+
+    def parse(self, response):
+        for point in get_points_list(response):
+            url = get_point_url(point)
+            yield response.follow(url, callback=build_point, headers=HEADERS)
+
+
+def get_points_list(response):
+    for point in response.json()["features"]:
+        if not point["properties"]["is_published"]:
+            log.info("Point not published: %s", point["properties"]["id"])
+            continue
+
+        yield point
+
+
+def get_point_url(point):
+    url = "https://www.mapotic.com/api/v1/maps/10392/public-pois/{}/"
+    return url.format(point["properties"]["id"])
+
+
+def build_point(response):
+    point = response.json()
+    lat, lng = point["point"]["coordinates"]
+    cat = CATEGORIES.get(point["category"]["name"]["en"], DEFAULT_CATEGORY)
+    return {
+        "country": "",
```

Comment:
Isn't country always Poland for this one?

Reply:
The largest amount of points is Czech, but there's several more countries, all have significant number of points.

Reply:
I see, ok

Reply:
There's a lot of meta info, but country and city are never to be found. Perhaps with the growing amount of incomplete geoinfo points we could do some offline geocoding?

E.g. https://github.com/richardpenman/reverse_geocode

Reply:
Yes, I'll handle the missing data and reverse geocoding in post-processing with `main.py`